api/v1alpha1/jobset_webhook.go

```diff
@@ -33,12 +34,20 @@ var _ webhook.Defaulter = &JobSet{}
 
 // Default implements webhook.Defaulter so a webhook will be registered for the type
 func (r *JobSet) Default() {
-	// Default job completion mode to indexed.
-	for i, rjob := range r.Spec.ReplicatedJobs {
-		if rjob.Template.Spec.CompletionMode == nil {
+	for i, _ := range r.Spec.ReplicatedJobs {
+		// Default job completion mode to indexed.
+		if r.Spec.ReplicatedJobs[i].Template.Spec.CompletionMode == nil {
 			r.Spec.ReplicatedJobs[i].Template.Spec.CompletionMode = completionModePtr(batchv1.IndexedCompletion)
 		}
+		// Enable DNS hostnames by default if job template uses indexed completion mode.
```

Comment:
We should always default to true, defaulting one parameter based on another is not a good practice.

Reply:
We cannot default to true unconditionally, because our validation enforces that enableDNSHostnames cannot be true if completionMode == NonIndexed.

Reply:
I think we can. The user should be setting the field to false explicitly if they set the completion mode to non-indexed.